spotbugs/etc/messages.xml

```diff
@@ -9128,6 +9137,51 @@ Using floating-point variables should not be used as loop counters, as they are
       ]]>
         </Details>
   </BugPattern>
+
+  <BugPattern type="ENV_USE_PROPERTY_INSTEAD_OF_ENV">
+    <ShortDescription>It is preferable to use portable Java property instead of environment variable.</ShortDescription>
+    <LongDescription>It is preferable to use portable Java property '{3}' instead of environment variable '{2}' in method {1}.</LongDescription>
+    <Details>
+      <![CDATA[<p>
+        Environment variables are not portable, the variable name itself (not only the value) may be different depending on the running OS.
+        Not only the names of the specific environment variables can differ (e.g. `USERNAME` in Windows and `USER` in Unix systems),
+        but even the semantics differ, e.g. the case sensitivity (Windows being case-insensitive and Unix case-sensitive).
+        Moreover, the Map of the environment variables returned by <code>java.lang.System.getenv()</code> and its collection views may not obey
+        the general contract of the <code>Object.equals(java.lang.Object)</code> and <code>Object.hashCode()</code> methods.
+        Consequently, using environment variables may have unintended side effects.
+        Also, The visibility of environment variables are also less restricted compared to Java Properties: they are visible to all descendants
```

Comment:
"Also, The" -> "Also, the"

"are also less restricted" -> "is less restricted"

Reply:
Thanks. Corrected them.